clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -141,10 +141,8 @@ public void ensureFreshMetadata() {
      * on the current poll if one is active, or the next poll.
      */
     public void wakeup() {
-        if (wakeupsEnabled) {
-            this.wakeup.set(true);
-            this.client.wakeup();
-        }
+        this.wakeup.set(true);
+        this.client.wakeup();
```

Comment:
I'm wondering if we also need to delay the call to `client.wakeup()`? For example, consider the following sequence:

1. disableWakeups()
2. wakeup()
3. poll(0)
4. enableWakeup()
5. poll(Long.MAX_VALUE)

The underlying wakeup on Selector would effect the `poll(0)`, but we would suppress the exception. Then there would be nothing to wake us up from the `poll(Long.MAX_VALUE)`.